File: domain-driven-design/python-web/src/domain/order/order_repository.py
# -*- coding: utf-8 -*-
from abc import ABC, abstractmethod

# 订单仓储接口（抽象类）
class OrderRepository(ABC):
    @abstractmethod
    def save(self, order):
        """保存订单"""
        pass

    @abstractmethod
    def find_by_id(self, order_id):
        """根据ID查找订单"""
        pass

    @abstractmethod
    def find_all(self, user_id):
        """查找所有订单"""
        pass

    @abstractmethod
    def delete(self, order_id):
        """删除订单"""
        pass

    @abstractmethod
    def find_by_customer_name(self, customer_name):
        """根据客户名称查找订单"""
        pass

File: practice-projects/springboot4ddd/src/main/java/com/github/microwind/springboot4ddd/domain/repository/OrderRepository.java
package com.github.microwind.springboot4ddd.domain.repository;

import com.github.microwind.springboot4ddd.domain.model.Order;

import java.util.List;
import java.util.Optional;

/**
 * 订单仓储接口
 *
 * @author jarry
 * @since 1.0.0
 */
public interface OrderRepository {

    /**
     * 保存订单
     */
    Order save(Order order);

    /**
     * 根据ID查找订单
     */
    Optional<Order> findById(Long id);

    /**
     * 根据订单号查找订单
     */
    Optional<Order> findByOrderNo(String orderNo);

    /**
     * 查找用户的所有订单
     */
    List<Order> findByUserId(Long userId);

    /**
     * 查找所有订单
     */
    List<Order> findAll();

    /**
     * 删除订单
     */
    void deleteById(Long id);
}
